lte/gateway/c/core/oai/tasks/amf/amf_app_state_manager.cpp

```diff
@@ -34,10 +34,71 @@ constexpr char AMF_GNB_UE_ID_AMF_UE_ID_TABLE_NAME[] =
 constexpr char AMF_TASK_NAME[]  = "AMF";
 const int NUM_MAX_UE_HTBL_LISTS = 6;
 
-/*hash function similar to default to initialize during hash table
- * initialization*/
-static hash_size_t amf_def_hashfunc(const uint64_t keyP) {
-  return (hash_size_t) keyP;
+/**
+ * When the process starts, initialize the in-memory AMF/NAS state and, if
+ * persist state flag is set, load it from the data store.
+ * This is only done by the amf_app task.
+ */
+int amf_nas_state_init(const amf_config_t* amf_config_p) {
+  return AmfNasStateManager::getInstance().initialize_state(amf_config_p);
+}
+
+/**
+ * Return pointer to the in-memory Amf/NAS state from state manager before
+ * processing any message. This is a thread safe call
+ * If the read_from_db flag is set to true, the state is loaded from data store
+ * before returning the pointer.
+ */
+amf_app_desc_t* get_amf_nas_state(bool read_from_redis) {
+  return AmfNasStateManager::getInstance().get_state(read_from_redis);
+}
+
+/**
+ * Write the AMF/NAS state to data store after processing any message. This is
+ * a thread safe call
+ */
+void put_amf_nas_state() {
+  magma5g::AmfNasStateManager::getInstance().write_state_to_db();
+}
+
+/**
+ * Release the memory allocated for the AMF NAS state, this does not clean the
+ * state persisted in data store
+ */
+void clear_amf_nas_state() {
+  AmfNasStateManager::getInstance().free_state();
+}
+
+map_uint64_ue_context_t get_amf_ue_state() {
+  return AmfNasStateManager::getInstance().get_ue_state_map();
+}
+
+void put_amf_ue_state(
```

Comment:
since this is not completed yet can we add this on the next PR when the UE state conversion is implemented?

Reply:
Comment Addressed